mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala

```diff
@@ -37,14 +51,10 @@ import org.apache.spark.sql.types.{DoubleType, StructField, StructType}
  */
 private[libsvm] class LibSVMRelation(val path: String, val numFeatures: Int, val vectorType: String)
     (@transient val sqlContext: SQLContext)
-  extends BaseRelation with TableScan with Logging with Serializable {
-
-  override def schema: StructType = StructType(
-    StructField("label", DoubleType, nullable = false) ::
-      StructField("features", new VectorUDT(), nullable = false) :: Nil
-  )
+  extends HadoopFsRelation with Logging with Serializable {
```

Comment:
Is it really necessary to mixin `Logging` trait here? `HadoopFsRelation` already does it.